python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,374 @@
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        lightning_config: Optional[Dict[str, Any]] = None,
+        *,
+        torch_config: Optional[TorchConfig] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        dataset_config: Optional[Dict[str, DatasetConfig]] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        datasets_iter_config: Optional[Dict[str, Any]] = None,
+        preprocessor: Optional[Preprocessor] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+    ):
+        train_loop_config = {
+            "lightning_config": lightning_config,
+            "datasets_iter_config": datasets_iter_config,
+        }
+
+        super(LightningTrainer, self).__init__(
+            train_loop_per_worker=_lightning_train_loop_per_worker,
+            train_loop_config=train_loop_config,
+            torch_config=torch_config,
+            scaling_config=scaling_config,
+            dataset_config=dataset_config,
+            run_config=run_config,
+            datasets=datasets,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+
+def _lightning_train_loop_per_worker(config):
+    """Per-worker training loop for a Lightning Trainer."""
+    if not config["lightning_config"]:
+        raise RuntimeError("'lightning_config' not specified in LightningTrainer!")
+
+    # Unpack all configs
+    ptl_config = config["lightning_config"]
+    datasets_iter_config = config["datasets_iter_config"]
+    trainer_config = ptl_config["_trainer_init_config"]
+    trainer_fit_params = ptl_config["_trainer_fit_params"]
+    module_class = ptl_config["_module_class"]
+    module_init_config = ptl_config["_module_init_config"]
+    ddp_strategy_config = ptl_config["_ddp_strategy_config"]
+
+    # Prepare data
+    datamodule = trainer_fit_params.get("datamodule", None)
+    train_dataloaders = trainer_fit_params.get("train_dataloaders", None)
+    val_dataloaders = trainer_fit_params.get("val_dataloaders", None)
+
+    train_ray_dataset = session.get_dataset_shard("train")
+    val_ray_dataset = session.get_dataset_shard("val")
+    if train_ray_dataset:
+        if datamodule:
+            logger.warning(
+                "Using Ray datasets as primary input. The 'datamodule' defined in "
+                "'LightningConfig.trainer_fit_params' will be ignored!"
+            )
+
+        datamodule = RayDataModule(
+            dataset_iter_config=datasets_iter_config,
+            train_dataset=train_ray_dataset,
+            val_dataset=val_ray_dataset,
+        )
+
+    # Prepare Lightning Module
+    lightning_module = module_class(**module_init_config)
+
+    # Prepare Lightning Trainer
+    trainer_config["enable_progress_bar"] = False
```

Comment:
Is this because Tune's output is too much for the native progress bar to work correctly?

Reply:
It's because Tune doesn't support progress bar well, the output will corrupted into tons of lines instead of one.

Reply:
can you please add this as a comment?

Reply:
ok